spring-boot-project/spring-boot-tools/spring-boot-buildpack-platform/src/main/java/org/springframework/boot/buildpack/platform/docker/transport/HttpClientTransport.java

```diff
@@ -146,14 +112,35 @@ private Response execute(HttpUriRequest request) {
 		try {
 			ClassicHttpResponse response = this.client.executeOpen(this.host, request, null);
 			int statusCode = response.getCode();
+
 			if (statusCode >= 400 && statusCode <= 500) {
 				byte[] content = readContent(response);
+				// Always close the response for error paths
 				response.close();
+
+				if (statusCode == 407) {
+					String detail = null;
+					Message json = deserializeMessage(content);
+					if (json != null && StringUtils.hasText(json.getMessage())) {
+						detail = json.getMessage();
+					}
+					else if (content != null && content.length > 0) {
+						detail = new String(content, StandardCharsets.UTF_8);
+					}
+
+					String msg = "Proxy authentication required for host: " + this.host.toHostString() + ", uri: "
+							+ request.getUri()
+							+ (StringUtils.hasText(detail) ? " - " + detail : "");
+
+					throw new ProxyAuthenticationException(msg);
```

Comment:
I'm still not sure that this is needed. Could it be a `DockerEngineException` with a better message?

Reply:
Please address this comment.